### DISCUSSION TITLE
Re-indexing after upgrading to 2026.06.6 worked for only one user.

### DISCUSSION THREAD

**Discussion by baskinsy** [Mailu/Mailu#3330] [User support] [answered]
## Environment & Version

- `Docker Compose version v2.28.1`
- Version: `2026.06.6`

## Description

Maybe this is an expected behavior but I am not sure. Feel free to close it if it is ok.

Following the post upgrade instructions to rebuild the dovecot indexes I got this:

```
root@mailu:/mailu# find /mailu/mail -type d -name xapian-indexes -prune -exec rm -r {} \+
root@mailu:/mailu# docker compose exec imap doveadm fts rescan -A
root@mailu:/mailu# docker compose exec imap doveadm user '*'|while read u; do echo "re-indexing $u";docker compose exec -T imap doveadm index -u $u '*'; done
re-indexing auser@example.com
```
Seems no other users indexes were recreated... 

## Replication Steps

Have tried the same commands numerous times with the same result.

## Observed behaviour

Only one user seems to be reindexed and everytime I run the commands a different one appears.

## Expected behaviour

I suppose all users should appear to be re-indexed

**Comment by ghostwheel42** [comment, marked as answer]
This is because the `docker compose exec -T imap doveadm index` is "eating" stadin.
This should work:
```
while read u; do echo "re-indexing $u"; docker compose exec -T imap doveadm index -u $u '*' </dev/null; done < <(docker compose exec imap doveadm user '*')
```

**Comment by baskinsy** [comment]
Thank you. Tried that and it is re-indexing all users now.
